src/Pacman/Player/index.js

```diff
@@ -75,26 +75,22 @@ export default class Player extends Component {
             }, ANIMATION_SPEED);
         }
 
-        if (this.props.onEnd) {
-            if(this.props.won) {
-                setImmediate(() => this.props.onEnd("WON"));
-            } else {
-                setImmediate(() => this.props.onEnd("LOST"));
-            }
+        if (this.props.onEnd && this.props.ended) {
+            setImmediate(() => this.props.onEnd(this.props.ended));
         }
 
         return null;
     }
     componentDidUpdate(prevProps) {
-        if ((!prevProps.lost && this.props.lost) || (!prevProps.won && this.props.won)) {
+        if (!prevProps.ended && (this.props.ended === 'WON' || this.props.ended === 'LOST')) {
```

Comment:
`'WON'` and `'LOST'` should be constants.

Reply:
The second condition can just be `this.props.ended`, because `ended` can only be either `WON`, `LOST` or falsey.